docs/management_api_v2.yml

```diff
@@ -0,0 +1,264 @@
+    description: Attribute descriptor.
+    type: object
+    required:
+      - name
+      - scope
+      - value
+    properties:
+      name:
+        type: string
+        description: |
+            A human readable, unique attribute ID, e.g. 'device_type', 'ip_addr', 'cpu_load', etc.
+      scope:
+        type: string
+        description: |
+            The scope of the attribute.
+
+            Scope is a string and acts as namespace for the attribute name.
+      description:
+        type: string
+        description: Attribute description.
+      value:
+        type: string
+        description: |
+            The current value of the attribute.
+
+            Attribute type is implicit, inferred from the JSON type.
+
+            Supported types: number, string, array of numbers, array of strings.
+            Mixed arrays are not allowed.
+    example:
+      application/json:
+        name: "cpu_load"
+        scope: "metrics"
+        description: "The current CPU load."
+        value: 42
+
+  Device:
+    type: object
+    properties:
+      id:
+        type: string
+        description: Mender-assigned unique ID.
+      updated_ts:
+        type: string
+        description: Timestamp of the most recent attribute update.
+      attributes:
+        type: array
+        items:
+          $ref: '#/definitions/Attribute'
+        description: A list of attribute descriptors.
+    example:
+      application/json:
+        id: "291ae0e5956c69c2267489213df4459d19ed48a806603def19d417d004a4b67e"
+        attributes:
+          - name: "ip_addr"
+            value: "1.2.3.4"
+            description: "IP address"
+          - name: "mac_addr"
+            value: "00.01:02:03:04:05"
+            description: "MAC address"
+          - name: "ports"
+            value:
+              - "8080"
+              - "8081"
+            description: "Open ports"
+        updated_ts: "2016-10-03T16:58:51.639Z"
+
+  Error:
+    description: Error descriptor.
+    type: object
+    properties:
+      error:
+        description: Description of the error.
+        type: string
+      request_id:
+        description: Request ID (same as in X-MEN-RequestID header).
+        type: string
+    example:
+      application/json:
+          error: "failed to decode device group data: JSON payload is empty"
+          request_id: "f7881e82-0492-49fb-b459-795654e7188a"
+
+  FilterPredicate:
+    description: Attribute filter predicate
+    type: object
+    required:
+      - scope
+      - attr
+      - type
+    properties:
+      scope:
+        type: string
+        description: |
+            The scope of the attribute.
+
+            Scope is a string and acts as namespace for the attribute name.
+      attribute:
+        type: string
+        description: |
+            Name of the attribute to be queried for filtering.
+      type:
+        type: string
+        description: Type or operator of the filter predicate.
+        enum: [$eq, $ne]
```

Comment:
we went back and forth with Eystein, but the final decision is to not support $ne for now.

(I know it's not a huge effort, but still).

Reply:
remove $ne